src/ome.ts

```diff
@@ -0,0 +1,226 @@
+            const field = (row * cols) + column;
+            if (source.endsWith('/')) {
+                source = source.slice(0, -1);
+            }
+            let imgSource = `${source}/${imagePaths[field]}`;
+            window.open(window.location.origin + '?source=' + imgSource);
+        }
+    }
+
+    return sourceData;
+}
+
+export async function loadOMEPlate(
+        config: ImageLayerConfig,
+        store: HTTPStore,
+        rootAttrs: RootAttrs,
+        acquisition: string | undefined
+    ): Promise<SourceData> {
+    const plateAttrs = rootAttrs.plate as OmePlateData;
+    if (!('columns' in plateAttrs) || !('rows' in plateAttrs)) {
+        throw Error(`Plate .zattrs missing columns or rows`);
+    }
+
+    let rows: number = plateAttrs.rows.length;
+    let columns: number = plateAttrs.columns.length;
+    let rowNames: string[] = plateAttrs.rows.map(row => row.name);
+    let columnNames: string[] = plateAttrs.columns.map(col => col.name);
+    let wellPaths = plateAttrs.wells.map(well => well.path + (well.path.endsWith('/') ? '' : '/'));
+
+    // Fields are by index and we assume at least 1 per Well
+    let field = '0';
+
+    // TEMP: try to support plates with plate/acquisition/row/column hierarchy
+    // Where plate.acquisitions = [{'path': '0'}]
+    let acquisitionPath = plateAttrs?.acquisitions?.[0]?.path;
+    acquisitionPath = acquisitionPath ? acquisitionPath + '/' : '';
+
+    // imagePaths covers whole plate (not sparse) - but some will be '' if no Well
+    const imagePaths = rowNames.flatMap(row => {
+        return columnNames.map(col => {
+            let wellPath = `${acquisitionPath}${row}/${col}/`;
+            return wellPaths.includes(wellPath) ? `${wellPath}${field}/` : '';
+        });
+    });
+
+    // Find first valid Image, loading each Well in turn...
+    let imageAttrs = undefined;
+    async function getImageAttrs(path: string): Promise<any> {
+        if (path === '') return
+        try {
+            return await getJson(store, `${path}.zattrs`);
+        } catch (err) {
+        }
+    }
+    for (let i = 0; i < imagePaths.length; i++) {
+        imageAttrs = await getImageAttrs(imagePaths[i]);
+        if (imageAttrs) {
+            break;
+        }
+    }
+
+    // Lowest resolution is the 'path' of the last 'dataset' from the first multiscales
+    let resolution = imageAttrs.multiscales[0].datasets.slice(-1)[0].path;
+
+    // Create loader for every Well. Some loaders may be undefined if Wells are missing.
+    async function createLoaderFromPath(path: string): Promise<ZarrLoader | undefined> {
+        try {
+            const data = await openArray({ store, path: `${path}${resolution}/` });
+            let l = createLoader([data]);
+            return l;
+        } catch (err) {
+            console.log(`Missing Well at ${path}`);
+        }
+    }
+    const promises = await pMap(imagePaths, createLoaderFromPath, { concurrency: 10 });
```

Comment:
perhaps add concurrency as a default arg to `loadOMEPlate`? Also there is a top-level `createLoaderFromPath` we can use here I think:

```suggestion
    const mapper = path => createLoaderFromPath(store, join(path, resolution));
    const promises = await pMap(imagePaths, mapper, { concurrency });
```

Reply:
Thanks - didn't think to use an extra `mapper` here.